caffeinated/admin/extend/messages/espresso_events_Messages_Hooks_Extend.class.php

```diff
@@ -161,14 +162,14 @@ public function messages_metabox($event, $callback_args)
                 ),
                 '<strong>',
                 '</strong>',
-                '<a href="' . $msg_activate_url . '">',
+                '<a href="' . esc_url($msg_activate_url) . '">',
```

Comment:
`esc_url()` can render a URL unusable and should only be used when displaying a URL (that doesn't need to work)

```suggestion
                '<a href="' . esc_url_raw($msg_activate_url) . '">',
```

Reply:
Fixed.